snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeTableOperations.java

```diff
@@ -0,0 +1,114 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+import java.sql.SQLException;
+import java.util.Map;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.jdbc.UncheckedInterruptedException;
+import org.apache.iceberg.jdbc.UncheckedSQLException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.snowflake.entities.SnowflakeTableMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class SnowflakeTableOperations extends BaseMetastoreTableOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeTableOperations.class);
+  private final String catalogName;
+
+  private final FileIO fileIO;
+  private final TableIdentifier tableIdentifier;
+
+  private final QueryFactory queryFactory;
+
+  private final Map<String, String> catalogProperties;
+
+  protected SnowflakeTableOperations(
+      QueryFactory queryFactory,
+      FileIO fileIO,
+      Map<String, String> properties,
+      String catalogName,
+      TableIdentifier tableIdentifier) {
+    this.queryFactory = queryFactory;
+    this.fileIO = fileIO;
+    this.catalogProperties = properties;
+    this.catalogName = catalogName;
+    this.tableIdentifier = tableIdentifier;
+  }
+
+  @Override
+  public void doRefresh() {
+    String location = null;
+    try {
+      LOG.debug("Getting metadata location for table {}", tableName());
+      location = getTableMetadataLocation();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e, "Interrupted during refresh");
+    } catch (SQLException e) {
+      // SQL exception happened when getting table from catalog
+      throw new UncheckedSQLException(
+          e, "Failed to get table %s from catalog %s", tableIdentifier, catalogName);
+    }
+
+    if (location == null || location.isEmpty()) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException(
+            "Failed to load table %s from catalog %s: dropped by another process",
```

Comment:
Do we know for sure that it was dropped by another process? Maybe we should just say "table could not be found".

Reply:
Looks like all this is vestigial; this can't happen in the current code because the metadata object itself is null for missing tables and we throw NoSuchTableException below for that case. Changed the location check to a precondition and removed this whole conditional branch.